pkg/logproto/logproto.proto

```diff
@@ -78,6 +78,9 @@ message QueryRequest {
   repeated string shards = 7 [(gogoproto.jsontag) = "shards,omitempty"];
   repeated Delete deletes = 8;
   Plan plan = 9 [(gogoproto.customtype) = "github.com/grafana/loki/v3/pkg/querier/plan.QueryPlan"];
+  // If populated, these represent the chunk references that the querier should
+  // use to fetch the data, plus any other chunks reported by ingesters.
+  ChunkRefGroup storeChunks = 10 [(gogoproto.jsontag) = "storeChunks"];
```

Comment:
nit: I think this would be better placed inside the Plan.
Would placing it there might the refactoring more difficult?